app/src/main/java/com/example/georgesamuel/dubaihotels/presentation/features/items/ListItemsViewModel.java

```diff
@@ -0,0 +1,65 @@
+package com.example.georgesamuel.dubaihotels.presentation.features.items;
+
+import androidx.lifecycle.MutableLiveData;
+import androidx.lifecycle.ViewModel;
+
+import com.example.georgesamuel.dubaihotels.entities.HotelsDetailsModel;
+import com.example.georgesamuel.dubaihotels.usecases.HotelsItemsUseCase;
+
+
+import io.reactivex.Observable;
+import io.reactivex.android.schedulers.AndroidSchedulers;
+import io.reactivex.disposables.CompositeDisposable;
+import io.reactivex.disposables.Disposable;
+import io.reactivex.observers.DisposableObserver;
+import io.reactivex.schedulers.Schedulers;
+
+public class ListItemsViewModel extends ViewModel {
+
+
+    MutableLiveData<HotelsDetailsModel> result;
+    MutableLiveData<Boolean> retrieving;
+    private HotelsItemsUseCase useCase;
+    private CompositeDisposable compositeDisposable;
+
+    public ListItemsViewModel() {
+        result=new MutableLiveData<>();
+        retrieving =new MutableLiveData<>();
+        useCase=new HotelsItemsUseCase();
+        compositeDisposable=new CompositeDisposable();
+    }
+
+        void getDetails(){
+
+        retrieving.setValue(true);
+       Observable<HotelsDetailsModel> call=useCase.getHotelsDetails();
+       Disposable disposable= call.subscribeOn(Schedulers.io())
```

Comment:
you can move subscribeOn and observeOn to Usecase

Reply:
Done